docs/reference/mapping/dynamic/templates.asciidoc

```diff
@@ -273,6 +273,12 @@ PUT my-index/_doc/1
 <4> The `ip_four` field is mapped as type `ip`.
 
 
+*Note:* when specifying patterns (for `match`, `unmatch`, etc.) in an array,
+{es} enforces that all values must be strings. However, for backwards
+compatibility, {es} allows non-string values (such as numbers) for
+single-valued match (and unmatch) fields.
```

Comment:
I wonder if we should document this. My worry is that users may end up relying on this behaviour, while it is there only to prevent breaking templates. We could consider as a follow-up deprecating non string single values. What do you think?

Reply:
Yeah, I agree. Users should probably never use anything except strings, so probably good not to encourage the other usage. I'll remove it.